drv/i2c-devices/src/pca9956b.rs

```diff
@@ -0,0 +1,311 @@
+
+    /// The LED index is too large
+    InvalidLED(u8),
+
+    /// Write buffer too large
+    WriteBufferTooLarge(usize),
+
+    /// Register is outside a valid auto-increment range
+    InvalidAutoIncReg(Register),
+}
+
+impl From<ResponseCode> for Error {
+    fn from(err: ResponseCode) -> Self {
+        Error::I2cError(err)
+    }
+}
+
+impl Pca9956B {
+    pub fn new(device: &I2cDevice) -> Self {
+        Self { device: *device }
+    }
+
+    /// Read a single Register
+    fn read_reg(&self, reg: Register) -> Result<u8, Error> {
+        self.device
+            .read_reg::<u8, u8>(reg as u8)
+            .map_err(Error::I2cError)
+    }
+
+    /// Read a number of Registers into `buf`
+    /// Note that CTRL_AUTO_INCR_MASK is set here to enable the device's
+    /// auto-increment feature, which has varying behaviors and limitations that
+    /// must be accounted for.
+    #[allow(dead_code)]
+    fn read_buffer(&self, reg: Register, buf: &mut [u8]) -> Result<(), Error> {
+        self.device
+            .read_reg_into((reg as u8) | CTRL_AUTO_INCR_MASK, buf)
+            .map_err(Error::I2cError)?;
+
+        Ok(())
+    }
+
+    /// Write a single Register
+    fn write_reg(&self, reg: Register, val: u8) -> Result<(), Error> {
+        let buffer = [reg as u8, val];
+        self.device.write(&buffer).map_err(Error::I2cError)
+    }
+
+    /// Write a number of Registers into `buf`
+    /// Note that CTRL_AUTO_INCR_MASK is set here to enable the device's
+    /// auto-increment feature, which has varying behaviors and limitations that
+    /// must be accounted for.
+    fn write_buffer(&self, reg: Register, buf: &[u8]) -> Result<(), Error> {
+        if buf.len() > MAX_BUF_SIZE {
+            return Err(Error::WriteBufferTooLarge(buf.len()));
+        } else if reg > MAX_AUTO_INC_REG {
+            return Err(Error::InvalidAutoIncReg(reg));
+        }
+
+        let mut data: [u8; MAX_BUF_SIZE + 1] = [0; MAX_BUF_SIZE + 1];
+        data[0] = (reg as u8) | CTRL_AUTO_INCR_MASK;
+        data[1..=buf.len()].copy_from_slice(buf);
+
+        self.device
+            .write(&data[..=buf.len()])
+            .map_err(Error::I2cError)
+    }
+
+    /// Sets the device's IREFALL register to `val`
+    /// IREFALL gets copied into all IREFn registers by the device.
+    pub fn set_iref_all(&self, val: u8) -> Result<(), Error> {
+        self.write_reg(Register::IREFALL, val)
+    }
+
+    /// Sets the device's PWMALL register to `val`
+    /// PWMALL gets copied into all PWMn registers by the device.
+    pub fn set_pwm_all(&self, val: u8) -> Result<(), Error> {
+        self.write_reg(Register::PWMALL, val)
+    }
+
+    /// Sets the PWMx register to `val` for a given `led`
+    pub fn set_a_led_pwm(&self, led: u8, val: u8) -> Result<(), Error> {
+        if led >= NUM_LEDS as u8 {
+            return Err(Error::InvalidLED(led));
+        }
+        let reg = FromPrimitive::from_u8((Register::PWM0 as u8) + led).unwrap();
```

Comment:
You can use `from_u8` on the `Register` type here, which makes it more obvious:
```suggestion
        let reg = Register::from_u8((Register::PWM0 as u8) + led).unwrap();
```